PWGCF/EbyEFluctuations/Tasks/nchCumulantsId.cxx

```diff
@@ -0,0 +1,671 @@
+    }
+    if (1.30 < track.p() // after 1.30 Pr and Ka lines of nSigma 3.0 will start intersecting
+        && std::abs(track.tpcNSigmaKa()) < 2.0 && std::abs(track.tofNSigmaKa()) < 2.0 && std::abs(track.tofNSigmaEl()) > 3.0 && std::abs(track.tofNSigmaPi()) > 3.0 && std::abs(track.tofNSigmaPr()) > 3.0 && std::abs(track.tofNSigmaDe()) > 3.0) {
+      return true;
+    }
+    return false;
+  }
+
+  // Proton
+  template <typename T>
+  bool selProtonTOF(T track)
+  {
+    if (track.p() <= 1.30 && std::abs(track.tpcNSigmaPr()) < 3.0 && std::abs(track.tofNSigmaPr()) < 3.0) {
+      return true;
+    }
+    if (1.30 < track.p() && track.p() <= 3.10                                                                                                                                                                                                     // after 1.30 Pr and Ka lines of nSigma 3.0 will start intersecting
+        && std::abs(track.tpcNSigmaPr()) < 3.0 && std::abs(track.tofNSigmaPr()) < 3.0 && std::abs(track.tofNSigmaEl()) > 3.0 && std::abs(track.tofNSigmaPi()) > 3.0 && std::abs(track.tofNSigmaKa()) > 3.0 && std::abs(track.tofNSigmaDe()) > 3.0 // Some Deuteron contamination is still coming in p dependent cuts
+    ) {
+      return true;
+    }
+    if (3.10 < track.p() // after 3.10 Pr and De lines of nSigma 3.0 will start intersecting
+        && std::abs(track.tpcNSigmaPr()) < 2.0 && std::abs(track.tofNSigmaPr()) < 2.0 && std::abs(track.tofNSigmaEl()) > 3.0 && std::abs(track.tofNSigmaPi()) > 3.0 && std::abs(track.tofNSigmaKa()) > 3.0 && std::abs(track.tofNSigmaDe()) > 3.0) {
+      return true;
+    }
+    return false;
+  }
+
+  // Deuteron
+  template <typename T>
+  bool selDeuteronTOF(T track)
+  {
+    if (track.p() <= 3.10 && std::abs(track.tpcNSigmaDe()) < 3.0 && std::abs(track.tofNSigmaDe()) < 3.0 && std::abs(track.tofNSigmaEl()) > 3.0 && std::abs(track.tofNSigmaPi()) > 3.0 && std::abs(track.tofNSigmaKa()) > 3.0 && std::abs(track.tofNSigmaPr()) > 3.0) {
+      return true;
+    }
+    if (3.10 < track.p() // after 3.10 De and Pr lines of nSigma 3.0 will start intersecting
+        && std::abs(track.tpcNSigmaDe()) < 2.0 && std::abs(track.tofNSigmaDe()) < 2.0 && std::abs(track.tofNSigmaEl()) > 3.0 && std::abs(track.tofNSigmaPi()) > 3.0 && std::abs(track.tofNSigmaKa()) > 3.0 && std::abs(track.tofNSigmaPr()) > 3.0) {
+      return true;
+    }
+    return false;
+  }
+
+  // Electron
+  template <typename T>
+  bool selElectronTOF(T track)
+  {
+    if (
+      (std::pow(track.tpcNSigmaEl(), 2) + std::pow(track.tofNSigmaEl(), 2)) < 9.00 && std::abs(track.tofNSigmaPi()) > 3.0 && std::abs(track.tofNSigmaKa()) > 3.0 && std::abs(track.tofNSigmaPr()) > 3.0 && std::abs(track.tofNSigmaDe()) > 3.0) {
+      return true;
+    }
+    return false;
+  }
+  //
+
+  // SelectionFunctions
+  // Pion
+  template <typename T>
+  bool selPion(T track, int& IdMethod)
+  {
+    if (!track.hasTOF() && selPionTPCInnerParam(track)) {
+      IdMethod = 0;
```

Comment:
I would suggest using named meaningful constants instead of plain numbers
It will improve the readability, maintenance and evolution of the code